File: backend/src/main/java/de/rainbowdev/backend/model/Post.java
package de.rainbowdev.backend.model;


import org.springframework.data.annotation.Id;

public record Post(
        @Id
        String id,
        String title,
        String userName,
        String postImg,
        String description,
        Boolean like,
        String url
) {
        public Post withId(String id) {
                return new Post (id, title, userName, postImg, description, like, url);
        }

        public Post withUrl(String url) {
                return new Post(id, title, userName, postImg, description, like, url);
        }
}

File: frontend/src/model/Post.ts
export type Post = {
    id: string,
    title: string,
    userName: string,
    postImg: string,
    description: string,
    profilePic: string,
    like: boolean,
    url: string
}

export type NewPost = {
    title: string,
    userName: string,
    postImg: string,
    description: string
}